dev-config-local-chart-extra-config.yaml

```diff
@@ -0,0 +1,26 @@
+# This config file is useful in the upgrade tests, where we upgrade from
+# either the latest stable chart or the latest dev release of the chart. This
+# config is only applied to the chart we upgrade to. It helps us handle
+# situations when we add new configuration options that would fail with a schema
+# validation error in the previous chart versions.
+#
+# Note that one could think that it would be possible to have dev-config.yaml
+# include this config and then pass --set hub.some-option=null to null it out
+# when it must not be passed, but that still triggers schema validation errors.
+#
+hub:
+  # FIXME: move loadRoles to dev-config.yaml after 2.0.0 is released.
+  loadRoles:
+    test-role-1:
+      description: Access to users' information and group membership
+      scopes: [users, groups]
+      users: [cyclops, gandalf]
+      services: [test]
+      groups: []
+    test-role-2-explicit-name:
+      name: test-role-2
+      description: Access to users' information and group membership
+      scopes: [users, groups]
+      users: [cyclops, gandalf]
+      services: [test]
+      groups: []
```

Comment:
Do you reckon we can test this in https://github.com/jupyterhub/zero-to-jupyterhub-k8s/blob/main/tests/test_hub.py by making a couple of API calls, one that should be allowed and one that should be blocked?

Reply:
I'd like to avoid ending up testing jupyterhub's own functionality, that is already tested. But, this is a z2jh config, so it makes sense to verify it is actually loading this configuration properly.

We also have the idle-culler configured and setup, it is currently failing with permission issues. Having a test to verify that doesn't fail is on my radar. #2407

Okay to settle for just testing that our idle-culler works for now?

Reply:
The idle-culler problems was why I thought a test would be good. If we can get a full idle-culler test (running, and also ideally test the culling) that would be great!

Reply:
This chart config isnt used by the idle culler, it directly adds to c.JuoyterHub.load_roles and services without doing anything here.

Hmmm, well i think perhaps doing a dedicated test is easier. Ill add a service and a key, then make a call to hub/api/info with a hub:read scoped token to be coupled with the service.

I note that to be failing currently for idle culler, so could be nice to verify separately as well.

Reply:
It's a change from what we've done in the past where most new config changes aren't explicitly tested, but I think it'd be good to try and do so in future, especially as the chart continues to grow in complexity. We could even use it as a guide for whether to accept a new feature- if it can't be tested in the chart then maybe it should be supported via a generic extension point rather than a config parameter?

Reply:
Im open to ideas on new tests, but the implementation idea matters a lot - are they complicated or easy to understand is a key thing. But we are out of scope for this PR to discuss that here.

---

@manics ive added a test to verify a token is granted access to an endpoint correctly.